util/src/kvdb.rs

```diff
@@ -417,11 +418,10 @@ impl Database {
 		};
 	}
 
-	/// Commit buffered changes to database.
-	pub fn flush(&self) -> Result<(), String> {
+	/// Commit buffered changes to database. Must be called under `flush_lock`
```

Comment:
It is an internal function. The code in the module only calls under the lock. I could add a `try_lock` runtime check but this seems an overkill considering we don't have such checks anywhere else in the codebase.